contracts/examples/Collateralized.sol

```diff
@@ -60,158 +70,309 @@ contract Collateralized is TermsContract {
         uint amount
     );
 
-    function Collateralized(address _debtRegistry) public {
+    function Collateralized(address _debtKernel, address _debtRegistry, address _tokenRegistry) public {
+        debtKernelAddress = _debtKernel;
         debtRegistry = DebtRegistry(_debtRegistry);
+        tokenRegistry = TokenRegistry(_tokenRegistry);
     }
 
-    function collateralize(
-        bytes32 agreementID,
-        address token,
-        uint amount,
-        uint lockupPeriodEndTimestamp
-    )
+    /**
+     * Abstract interface for a function that returns the last timestamp
+     * at which any repayment is expected -- i.e., the debt agreement's
+     * term's ending.
+     *
+     * @param agreementId bytes32 The debt agreement's ID
+     * @return _timestamp uint
+     */
+    function getTermEndTimestamp(
+        bytes32 agreementId
+    ) public view returns (uint _timestamp);
+
+    /**
+     * Hook that is called by the DebtKernel upon a debt agreement's
+     * issuance.  Parses out collateralization-specific parameters
+     * from the debt agreement's terms contract parameters and
+     * pulls collateral from the debtor based on those parameters.
+     *
+     * @param agreementId bytes32 The debt's agreement ID
+     * @param collateralizer address The debt agreement's collateralizer
+     * @return _success bool
+     */
+    function registerTermStart(bytes32 agreementId, address collateralizer)
         public
+        returns (bool _success)
     {
-        // the amount being put up for collateral must exceed 0.
-        require(amount > 0);
+        require(msg.sender == debtKernelAddress);
+
+        address collateralToken;
+        uint collateralAmount;
+        uint gracePeriodInDays;
 
-        // the lockup period must occur at some point in the future.
-        require(lockupPeriodEndTimestamp > block.timestamp);
+        (collateralToken, collateralAmount, gracePeriodInDays) = retrieveCollateralParameters(agreementId);
+
+        // the amount being put up for collateral must exceed 0.
```

Comment:
probably an unnecessary comment, since that's exactly what the code says below

Reply:
Good point, fixing.